gnowledge/gstudio
# ProgramEvent, CourseEvent, Notifications in workflow

ProgramEventGroup(PE) and CourseEventGroup(CE) are similar to ModeratingGroup.
Both PE and CE have:
  1. a new create/edit template.
  2. curation-workflow
  3. subgroups have their respective member_of

[A]. ProgramEvent
PE can be created from any group's dashboard, a button 'New ProgramEvent' will appear in Actions section. This button will now be visible only to SuperUser.

[B]. CourseEvent
Create a Course and in its detail template, 'Announce' button will be used to create CE

[C]. Notifications

1. When a user uploads file to editable_moderated group:
      - email notification to contributor with link of resource's moderation-status (in user-group)
      - task to immediate mod groups' admin

2. On Approving Resource
     - if resource is approved in curation group, i.e resource will be published to top_group, existing task 's status changed to 'Feedback', assignee will be contributor and top_group's admin, link of approved resource(in top_group)
     - if resource is approved in clearing house group, existing tasks' group_set will be changed to next mod group, assignee will be next mod group's admin and link of resource(in mod_group)


What to test:
1. Create Editable_moderated group and test workflow
2. Create ProgramEvent group and test workflow
3. Create CourseEvent group and test workflow
4. Test for 'Join' button enable/disable under PE, based on enrollment_end date.
5. Check for notifications.

RTs.csv will be sent on mailing-list which requires RT 'has_current_approval_task'  for linking the approval_task and uploaded_resource




==========
Updates in Notification:
	
	On Final publish. DELETE task and send mail to contributor and top group admin.

	When a resource is moved from one mod group to other:
		send notification to contributor about update.

	notifications in try-catch


PE/CE:
	
	Submit button in create_event_group	, label changed
	
	hidden edit policy and level of moderation in create_event_group.html
